src/onnx/onnx.cpp

```diff
@@ -2424,11 +2426,22 @@ struct onnx_parser
         auto l = add_broadcastable_binary_op(args[0], args[1], "equal");
         if(l->get_shape().type() != shape::bool_type)
         {
-            l = prog.add_instruction(op::convert{shape::bool_type}, l);
+            l = prog.add_instruction(make_op("convert", {{"target_type", "bool_type"}}), l);
         }
         return l;
     }
 
+    instruction_ref
+    parse_where(const std::string&, const node_info&, std::vector<instruction_ref> args)
+    {
+        auto type = args[1]->get_shape().type();
+        auto cond = prog.add_instruction(
+            make_op("convert", {{"target_type", shape::type_string(type)}}), args[0]);
+        auto diff = add_broadcastable_binary_op(args[1], args[2], "sub");
```

Comment:
can we make a comment in the code of the equation being used that is equivalent to the conditional statement?

Reply:
add one